### PR TITLE
[test] fix restartServerByDeletingSSTFiles tests with checksum enabled

### PR DESCRIPTION
**Pull request by m-nagarajan**
## Summary
This test originally had the following flow:
* Delete the SST files.
* Restart the servers to simulate a crash and restart in the midst of SST file move.

However, this flow introduced an issue when checksum was enabled:
Graceful shutdown of the server after SST file deletion triggers a final offset sync, which throws an exception when files are missing during checksum calculations.

Modified the test to:
Stop the server -> Delete the SST files -> Start the server.

## How was this PR tested?
GH CI

## Does this PR introduce any user-facing changes?
- [x] No. You can skip the rest of this section.
- [] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Review by adamxchen — COMMENTED** [review]
> Graceful shutdown of the server after SST file deletion triggers a final offset sync, which throws an exception when files are missing during checksum calculations.

Can you add one more test (either a unit test or integ test) to test this path?

**Comment by sixpluszero** [comment]
> > Graceful shutdown of the server after SST file deletion triggers a final offset sync, which throws an exception when files are missing during checksum calculations.
> 
> Can you add one more test (either a unit test or integ test) to test this path?

I think the change itself is good. Regarding the suggestion by Adam, I think it is OK to have it to validate the final sync() action as you mentioned.

Nit: Is it possible to merge the non-aa test with the A/A test with some test parameters? Not necessarily need to happen in this PR if you think it is complicated but I suppose it can just be turned off by setting store to A/A = false?

**Comment by m-nagarajan** [comment]
@adamxchen @sixpluszero added more unit tests, please take another look.

File: clients/da-vinci-client/src/test/java/com/linkedin/davinci/store/rocksdb/RocksDBSstFileWriterTest.java [hunk 1]
```diff
@@ -255,6 +259,124 @@ public void testOpenWithAllValidFiles() throws IOException {
...
+      // create 10 sst file
+      createSstFiles(10);
+
+      rocksDBSstFileWriter.open(checkpointedInfo, Optional.of(() -> "1".getBytes()));
+      // check that only 7 files exists (0-5 after cleanup by checkDatabaseIntegrity)
+      // and 6 after rocksDBSstFileWriter.open opens a new file
+      Assert.assertEquals(getNumberOfFilesInTempDirectory(), 7);
+      rocksDBSstFileWriter.put("key".getBytes(), ByteBuffer.wrap("value".getBytes()));
+      deleteAllSstFiles(getNumberOfFilesInTempDirectory());
+      // call sync after deleting all the files to mimic exception thrown during graceful shutdown of servers with
```
Comment (sixpluszero) [inline clients/da-vinci-client/src/test/java/com/linkedin/davinci/store/rocksdb/RocksDBSstFileWriterTest.java:377]: Other change looks good, but I am confused by the intention of this particular test. Could you explain a bit more about this one?

Comment (m-nagarajan) [reply clients/da-vinci-client/src/test/java/com/linkedin/davinci/store/rocksdb/RocksDBSstFileWriterTest.java:377]: This was the RC behind this test failing after enabling checksum calculation. so added a test to verify that the checksum fails if there are no sst files during the sync() which is expected.

**Review by sixpluszero — APPROVED** [review]
Thanks for the fix!

**Pull request merged**